app/controllers/saml_idp_controller.rb

```diff
@@ -51,12 +51,6 @@ def confirm_user_is_authenticated_with_fresh_mfa
     redirect_to user_two_factor_authentication_url if remember_device_expired_for_sp?
   end
 
-  def saml_request_valid?(saml_request)
```

Comment:
This seemed too juicy to be true.  I dug into the saml gem and it appears to be ok to remove.